tools/MINGW-packages/mingw-w64-imagemagick/fix-compiler-errors.patch

```diff
@@ -0,0 +1,39 @@
+diff -ru a/magick/utility-private.h b/magick/utility-private.h
+--- a/magick/utility-private.h	2024-05-16 16:18:24.863994300 -0700
++++ b/magick/utility-private.h	2024-05-16 16:18:57.352206600 -0700
+@@ -273,7 +273,7 @@
+    path_wide=create_wchar_path(path);
+    if (path_wide == (WCHAR *) NULL)
+      return(-1);
+-   status=wstat(path_wide,attributes);
++   status=_wstati64(path_wide,attributes);
+    path_wide=(WCHAR *) RelinquishMagickMemory(path_wide);
+    return(status);
+ #endif
+
+diff -ru a/magick/utility.c b/magick/utility.c
+--- a/magick/utility.c  2020-05-31 09:04:03.000000000 -0700
++++ b/magick/utility.c 2024-05-17 08:05:06.523764900 -0700
+@@ -67,6 +67,9 @@
+ #if defined(MAGICKCORE_HAVE_PROCESS_H)
+ #include <process.h>
+ #endif
++#if defined(MAGICKCORE_HAVE_GETPAGESIZE)
++int getpagesize();
```

Comment:
FTR I'm not proud of this, but I couldn't figure out how configure was determining that this function was available. I couldn't find it in any of the msys2 headers.  The new GCC was just complaining about the function signature not being defined, so I forward declared it. The code linked fine so this appears to be good enough to get things building. :/ Upstream Imagemagick code has changed significantly in this area so I just went with this instead of trying to backport the upstream changes to make the compiler happy.

Reply:
Update IM6 (we do not want v7)  or cherry-pick patches? (https://github.com/ImageMagick/ImageMagick6/commits/main/)

I can look into the IM stuff at a later date.

Reply:
Is it ok if I update the IM6 package in a followup PR? It does look time some of the fixes I've made are on upstream now and I think I could actually clean out some of the older patches in our code if I update to HEAD on the IM6 main. I'm just trying to get stuff building again with minimal other changes here. It looks like IM6 releases every 2 weeks or so I'd expect an official release in the next week or so and I'm happy to do the work to update to that when it  happens.

Reply:
Update of IM6 is ok as long as it does not break the unit tests (https://github.com/NatronGitHub/Natron-Tests).

Reply:
ok. Do these actually pass still? I tried running these a few times in the past and got all sorts of failed tests. Are these supposed to work on all platforms or are they primarily run on one in particular? I've only tried on Windows, but I'm curious if these are still passing on Linux or Mac. I haven't tried Linux lately, and don't have a Mac to try them on.

Reply:
Windows builds should be identical (or better) than tests from 2.5.0:
* https://github.com/NatronGitHub/Natron/releases/download/v2.5.0/Natron-2.5.0-Windows-x86_64-tests.txt
* https://github.com/NatronGitHub/Natron/releases/download/v2.5.0/Natron-2.5.0-Windows-x86_64-unit_tests_failures.zip

Any change to OIIO, OCIO, IM etc must pass the unit tests. If they now pass I don't know, but they have to before any release.

For now just focus on Windows, we will take care of Linux/macOS later.